### PR TITLE
Add ibi resources

### PR DESCRIPTION
**Pull request by trewest**
Add imageclusterinstall resource and APIs

### CODE REVIEW

File: pkg/ibi/imageclusterinstall.go [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+	}, imageClusterInstall)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return imageClusterInstall, err
+}
+
+// Create generates a imageclusterinstall on the cluster.
```
Comment (achuzhoy) [inline pkg/ibi/imageclusterinstall.go:365]: nit: an

File: pkg/ibi/imageclusterinstall_test.go [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+		}
+	}
+}
+func TestImageClusterInstallWithVersion(t *testing.T) {
+	testCases := []struct {
+		version          string
+		expectedErrorMsg string
+	}{
+		{
+			version:          "mysecretsshkey",
```
Comment (achuzhoy) [inline pkg/ibi/imageclusterinstall_test.go:320]: nit: maybe change the string

File: pkg/ibi/imageclusterinstall_test.go [hunk 3]
```diff
@@ -0,0 +1,772 @@
...
+		}
+	}
+}
+func TestImageClusterInstallGetCompletedCondition(t *testing.T) {
+	testCases := []struct {
+		status                   ibiv1alpha1.ImageClusterInstallStatus
+		expectedConditionMessage string
+		expectedError            error
+	}{
+		{
```
Comment (achuzhoy) [inline pkg/ibi/imageclusterinstall_test.go:346]: Do we want another testcase with wrong status?

**Review by achuzhoy — DISMISSED** [review]

File: pkg/ibi/imageclusterinstall_test.go (hunk 3 above)
Comment (trewest) [reply pkg/ibi/imageclusterinstall_test.go:346]: The method just returns a condition object based on the type of the condition. I have test cases for when the condition type appears and doesn't appear in the status.

File: pkg/ibi/imageclusterinstall.go [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+		return builder
+	}
+
+	builder.Definition.Spec.Hostname = hostname
+
+	return builder
+}
+
+// WithClusterDeployment links imageclusterinstall to an existing cluster deployment.
+func (builder *ImageClusterInstallBuilder) WithClusterDeployment(name string) *ImageClusterInstallBuilder {
```
Comment (kononovn) [inline pkg/ibi/imageclusterinstall.go:154]: Can you re-name arg to clusterDeploymentName ?

File: pkg/ibi/imageclusterinstall.go [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+		builder.Definition.Spec.ClusterDeploymentRef = &corev1.LocalObjectReference{}
+	}
+
+	builder.Definition.Spec.ClusterDeploymentRef.Name = name
+
+	return builder
+}
+
+// WithExtraManifests includes manifests via configmap name.
+func (builder *ImageClusterInstallBuilder) WithExtraManifests(name string) *ImageClusterInstallBuilder {
```
Comment (kononovn) [inline pkg/ibi/imageclusterinstall.go:177]: Can you re-name arg to extraManifestsName ?

File: pkg/ibi/imageclusterinstall.go [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+		return builder
+	}
+
+	builder.Definition.Spec.Version = version
+
+	return builder
+}
+
+// GetCompletedCondition returns Completed condition from imageclusterinstall.
+func (builder *ImageClusterInstallBuilder) GetCompletedCondition() (*hivev1.ClusterInstallCondition, error) {
```
Comment (kononovn) [inline pkg/ibi/imageclusterinstall.go:256]: I think you can simplify  next 4 function by adding getCondition function which accepts condition argument. 
 GetCompletedCondition,GetFailedCondition,GetRequirementsMetCondition,GetStoppedCondition can re-use getCondition with different argument. WDYT ?

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments above.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**